src/sentry/integrations/jira/search.py

```diff
@@ -24,11 +30,7 @@ def _get_formatted_user(self, user):
 
     def get(self, request, organization, integration_id):
         try:
-            integration = Integration.objects.get(
-                organizations=organization,
-                id=integration_id,
-                provider='jira',
-            )
+            integration = self._get_integration(organization, integration_id)
```

Comment:
Your solution works perfectly fine. But when I was looking at this the only thing that appears to change here is the provider name. Putting the provider name in a class variable probably is all you need here . So something like:
```python
class JiraSearchEndpoint(IntegrationEndpoint):
    provider = 'jira'
    def get(self, request, organization, integration_id):
           try:
                   integration = Integration.objects.get(
                           organizations=organization,
                           id=integration_id,
                           provider=self.provider,
                   )
```
So then JiraServer is more or less the same 
```python
class JiraServerSearchEndpoint(JiraSearchEndpoint):
    provider = 'jira-server'
```